src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -363,7 +367,7 @@ private BoundExpression MakeEqual(BoundExpression loweredLiteral, BoundExpressio
                     BoundExpression input = _tempAllocator.GetTemp(test.Input);
                     BoundExpression output = _tempAllocator.GetTemp(new BoundDagTemp(evaluation.Syntax, typeEvaluation1.Type, evaluation));
                     sideEffect = _factory.AssignmentExpression(output, _factory.As(input, typeEvaluation1.Type));
-                    testExpression = _factory.ObjectNotEqual(output, _factory.Null(output.Type));
+                    testExpression = _factory.ObjectNotEqual(output, _factory.Null(output.Type!));
```

Comment:
assert?

Reply:
I wish there were a way of expressing that the BoundExpression returned from GetTemp always has a non-null type.

---
In reply to: [387364151](https://github.com/dotnet/roslyn/pull/42068#discussion_r387364151) [](ancestors = 387364151)

Reply:
FWIW, We have an issue in nullable tracking this. We brainstormed an extension of NotNull attributes that would allow listing members that are not null on a parameter or returned value.

---
In reply to: [391779259](https://github.com/dotnet/roslyn/pull/42068#discussion_r391779259) [](ancestors = 391779259,387364151)